src/pages/organisationUnits/list/OrganisationUnitList.tsx

```diff
@@ -167,35 +140,45 @@ export const OrganisationUnitList = () => {
 
     const computedRoot = useMemo(() => {
         return flatOrgUnits.filter(
-            (ou) => !ou.parent || userRootOrgUnitIds.includes(ou.id)
+            (ou) => !ou.parent || initialExpandedState[ou.id]
         )
-    }, [flatOrgUnits, userRootOrgUnitIds])
+    }, [flatOrgUnits, initialExpandedState])
 
-    // handle when expanded
     const handleExpand = useCallback(
         (valueOrUpdater: Updater<ExpandedState>) => {
-            // we are just using ExpandedStateList in state, because we need the exact Ids to load
-            // but API exposes ExpandedState (includes true to expand all)
-            // so we handle and translate that to expand all loaded units
-            const setExpandedFunc = isFiltering
-                ? setExpandedDuringFilter
-                : setExpanded
-            const expandAll = () =>
-                Object.fromEntries(
-                    flatOrgUnits.map((ou) => [ou.id, true] as const)
-                )
+            // when we expand something and are not filtering, we need to load the children
+            // also translate expandedState === true (expand all) to expand all loaded units
+            const getAllExpanded = () =>
```

Comment:
We may not actually want this functionality, as I think it's generally a bad idea. Currently no way to trigger this from UI.